File: Introductory Problems/Trailing Zeros/trailing_zeros.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

void solve(){
	ll n, ans = 0, deno = 5;
	cin>>n;
	while(n>=deno){
		ans += n/deno;
		deno *= 5;
	}
	cout<<ans;
	return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}

File: Introductory Problems/Trailing Zeros/trailingZeros.java
import java.util.Scanner;

public class trailingZeros {
    private static void solve(){
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextInt(), ans = 0, deno = 5;
        while(n >= deno){
            ans = ans + (n/deno);
            deno = deno*5;
        }
        System.out.println(ans);
        scanner.close();
        return;
    }
    
    public static void main(String args[]){
        solve();
    }
}
